src/main/java/controller/user/SignInServlet.java

```diff
@@ -0,0 +1,54 @@
+package controller.user;
+
+import factory.UserServiceFactory;
+import model.User;
+import service.user.UserService;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/signIn")
+public class SignInServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("users.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String email = req.getParameter("email");
+        String password = req.getParameter("password");
+
+        if (email.isEmpty() || password.isEmpty()) {
+            if (email.isEmpty()) {
+                req.setAttribute("error", "Email can not be empty! Try another.");
+            } else {
+                req.setAttribute("email", email);
+            }
+            if (password.isEmpty()) {
+                req.setAttribute("error", "Password can not be empty! Try another.");
+            } else {
+                req.setAttribute("password", password);
+            }
+        }
+
+        for (User user : userService.getAll()) {
```

Comment:
better create separate method for this at service layer

Reply:
rebuilt